polylogue/local_sync.py

```
@@ -285,35 +328,96 @@ def _process_single(session_path: Path) -> Dict[str, object]:
+                _maybe_plain_progress(done, len(session_list))
+        else:
+            results_by_index: List[Dict[str, object]] = [{} for _ in session_list]
+            with ThreadPoolExecutor(max_workers=jobs) as executor:
+                future_map = {executor.submit(_process_single, session_path): idx for idx, session_path in enumerate(session_list)}
+                done = 0
+                for future in as_completed(future_map):
+                    idx = future_map[future]
+                    session_path = session_list[idx]
+                    try:
+                        results_by_index[idx] = future.result()
+                    except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)